src/main/scala/mesosphere/marathon/api/v2/AppHelpers.scala

```diff
@@ -52,15 +53,15 @@ object AppHelpers {
 
   private def checkAuthorization[A, B >: A](action: AuthorizedAction[B], resource: A)(implicit identity: Identity, authorizer: Authorizer): A = {
     if (authorizer.isAuthorized(identity, action, resource)) resource
-    else throw AccessDeniedException()
+    else throw RejectionException(Rejection.AccessDeniedRejection(authorizer, identity))
```

Comment:
What is the motivation for throwing a rejection exception instead of the `AccessDeniedException`? Couldn't `AccessDeniedException` just extend `RejectionException`?

Reply:
I'm probably missing something. Currently everything is an exception. We can go for rejections but I would do this in a second pull request. This one is fairly big and introduces multiple changes. If it would introduce `sendResponse` and `authenticatedAsync` only it would be easier to follow. The second step could be the introduction of rejections.